### DISCUSSION TITLE
Setting CONFIG_SYS_CLOCK_TICKS_PER_SEC to 1 million in a tickless build

### DISCUSSION THREAD

**Discussion by pavel-kirienko** [zephyrproject-rtos/zephyr#72487] [Q&A]
Is it a sensible idea to set `CONFIG_SYS_CLOCK_TICKS_PER_SEC = 1000000` in a tickless kernel in order to increase the time resolution, or are there side effects one needs to be aware of? I am obviously aware of the more frequent timer overflows; this is not an issue in my case. Same goes for the increased power consumption by the timer.

Thanks!

**Comment by ndrs-pst** [comment]
@pavel-kirienko I think the downside would be the maximum time the kernel can wait (`MAX_TICKS`), which depends on the timer being used as the system tick. Digging into `kernel/timeout.c` might give you more insights. 😃

---

https://github.com/zephyrproject-rtos/zephyr/blob/012520d2d7e8d2bbc914bb5735bf1563abd0e4ea/drivers/timer/cortex_m_systick.c#L15-L21

https://github.com/zephyrproject-rtos/zephyr/blob/012520d2d7e8d2bbc914bb5735bf1563abd0e4ea/drivers/timer/esp32_sys_timer.c#L22-L26
